arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedColumnIterator.java

```diff
@@ -69,12 +69,20 @@ public boolean producesDictionaryEncodedVector() {
   }
 
   public abstract class BatchReader {
-    public void nextBatch(FieldVector fieldVector, int typeWidth, NullabilityHolder holder) {
+    public void nextBatch(
+        int numValsToRead, FieldVector fieldVector, int typeWidth, NullabilityHolder holder) {
       int rowsReadSoFar = 0;
-      while (rowsReadSoFar < batchSize && hasNext()) {
+      while (rowsReadSoFar < batchSize && hasNext() && rowsReadSoFar < numValsToRead) {
         advance();
+        int expectedBatchSize;
+        if (numValsToRead < 0) {
+          expectedBatchSize = batchSize - rowsReadSoFar;
```

Comment:
[doubt] Considering rowsReadSoFar = 0 and numValsToRead < 0 ? in this case will we ever enter the while loop ? how are we keep the behaviour without the feature intact ?

Reply:
Technically, I think this `numValsToRead < 0` should not happen, maybe just throw Exception?
```
        if (numValsToRead < 0) {
            throw new IllegalArgumentException("numValsToRead has invalid value");
        } else {
          expectedBatchSize = Math.min(batchSize - rowsReadSoFar, numValsToRead - rowsReadSoFar);
        }
```

Reply:
Should we throw illegal state instead ?